prometheus/summary.go

```diff
@@ -181,7 +181,8 @@ func NewSummary(opts SummaryOpts) Summary {
 
 func newSummary(desc *Desc, opts SummaryOpts, labelValues ...string) Summary {
 	if len(desc.variableLabels) != len(labelValues) {
-		panic(errInconsistentCardinality)
+		err := fmt.Errorf("%s: labels and values for '%s' don't match: labels %v, values %v", errInconsistentCardinality, desc.fqName, desc.variableLabels, labelValues)
+		panic(err)
```

Comment:
See above for both lines. (As we have this four times, you might consider a helper function for creating the error, and then just write `panic(makeInconsistentCardinalityError(desc.variableLabels, labelValues))`

Reply:
Created the function in `labels.go`